onnxruntime/core/framework/execution_frame.cc

```diff
@@ -254,8 +256,8 @@ ExecutionFrame::ExecutionFrame(const std::vector<int>& feed_mlvalue_idxs, const
             // it's less efficient (the arena will add some overhead to coalesce individual allocations
             // back into blocks on 'free'), but better than failing completely.
             try {
-              buffer = alloc->Alloc(mem_patterns_->patterns[i].PeakSize());
-
+              static_activation_memory_in_bytes_ = mem_patterns_->patterns[i].PeakSize();
```

Comment:
>static_activation_memory_in_bytes_ [](start = 14, length = 34)

could there by multiple patterns? i.e. CPU pinned, CUDA, etc.

Reply:
yeah, I was wondering the same case -- should we use += here?

---
In reply to: [440499658](https://github.com/microsoft/onnxruntime/pull/4241#discussion_r440499658) [](ancestors = 440499658)

Reply:
There are multiple patterns. I will split it and therefore avoid `+=`.

---
In reply to: [440501380](https://github.com/microsoft/onnxruntime/pull/4241#discussion_r440501380) [](ancestors = 440501380,440499658)